Core/Generators/Rust/RustGenerator.cs

```diff
@@ -837,6 +1004,72 @@ private static void WriteRecordImpl(IndentedStringBuilder builder, string name,
             }
         }
 
+
+        private void WriteFromBorrowedToOwnedForFields(IndentedStringBuilder builder, IEnumerable<Field> fields,
+            bool optionalFields = false, string obj = "Self", string attrPrefix = "value.") =>
+            builder.CodeBlock(obj, _tab, () =>
+            {
+                foreach (var f in fields)
+                {
+                    var atrName = MakeAttrIdent(f.Name);
+                    var intoMethod = BorrowedIntoOwnedMethod(f.Type);
+                    if (optionalFields && intoMethod != "")
+                    {
+                        // we have to map because it is an optional and we can't just move
+                        builder.AppendLine($"{atrName}: {attrPrefix}{atrName}.map(|value| value{intoMethod}),");
+                    }
+                    else
+                    {
+                        // we don't need to map, but we do need to convert
+                        builder.AppendLine($"{atrName}: {attrPrefix}{atrName}{intoMethod},");
+                    }
+                }
+            });
+
+        /// <summary>
+        /// Generate the `.into()` or equivalent for converting from a borrowed type to an owned type.
+        /// </summary>
+        private string BorrowedIntoOwnedMethod(TypeBase type)
```

Comment:
This was probably the hardest part to get right

Reply:
The logic looks sound to me, at first glance! :+1: